sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs

```diff
@@ -15,6 +16,9 @@ namespace Azure.Core.Json
     /// <summary>
     /// A mutable representation of a JSON value.
     /// </summary>
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+    [RequiresUnreferencedCode("This class utilizes reflection-based JSON serialization and deserialization which is not compatible with trimming.")]
```

Comment:
Same question here as https://github.com/Azure/azure-sdk-for-net/pull/37734/files#r1324929628

Reply:
There are only 2 warnings coming from this class specifically, the problem is with `RootElement` since MutableJsonElement isn't compatible with trimming as a whole.

[L44](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs#L44) /[L131](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs#L131):
```csharp
42        public MutableJsonElement RootElement
43        {
44            get => new(this, _originalDocument.RootElement, string.Empty);
45        }
``` 
```csharp
120        public void WriteTo(Utf8JsonWriter writer)
121 #pragma warning restore AZC0014 // Avoid using banned types in public API
122        {
...
130
131            RootElement.WriteTo(writer);
```

If it was just those two we could just annotate the getter and the WriteTo method but the reason the whole class has to be annotated is because of the `MutableJsonDocumentConverter`:
[L222](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs#L222):
```csharp
213         private class MutableJsonDocumentConverter : JsonConverter<MutableJsonDocument>
214         {
215            public override MutableJsonDocument Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
216            {
217                return Parse(ref reader);
218            }
219
220            public override void Write(Utf8JsonWriter writer, MutableJsonDocument value, JsonSerializerOptions options)
221            {
222                value.WriteTo(writer);
223            }
224        }
```
`Write` can't be annotated because in order for a method to be annotated, all base/inherited methods have to be annotated with the same annotation. The fix for that is to annotate the whole class. If we annotate `MutableJsonConverter`, then it creates a warning on `MutableJsonDocument` because of it's usage in the attribute. The only way to fix that warning is to annotate the whole class.